SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -136,6 +144,102 @@ void FLocalDeploymentManager::RefreshServiceStatus()
+	bool bSuccess = false;
+	// Set our broadcast address
+	TSharedRef<FInternetAddr> BroadcastAddr = SocketSubsystem->CreateInternetAddr();
+	BroadcastAddr->SetBroadcastAddress();
+	BroadcastAddr->SetPort(Port);
+	// Now the listen address
+	TSharedRef<FInternetAddr> ListenAddr = SocketSubsystem->GetLocalBindAddr(*GLog);
+	ListenAddr->SetPort(Port);
+	// Now create and set up our sockets
+	FSocket* ListenSocket = SocketSubsystem->CreateSocket(NAME_Stream, TEXT("LAN beacon"), false);
+	if (ListenSocket != NULL)
+	{
+		ListenSocket->SetReuseAddr();
+		ListenSocket->SetNonBlocking();
+		ListenSocket->SetRecvErr();
+		// Bind to our listen port
+		if (ListenSocket->Bind(*ListenAddr))
+		{
+			bSuccess = ListenSocket->Listen(30);
+		}
+		else
+		{
+			UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to bind listen socket to addr (%s) for LAN beacon"), *ListenAddr->ToString(true));
+		}
+	}
+	else
+	{
+		UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to create listen socket for LAN beacon"));
+	}
+	bool bSuccessfullyBoundPort = bSuccess && ListenSocket;
+	if (ListenSocket != NULL) {
+		// Kill the blocking thing we just opened.
+		ListenSocket->Close();
+	}
+	/* Either we couldn't create the socket or couldn't listen on it so the port is probably not bound */
+	return !bSuccessfullyBoundPort;
+}
+
+bool FLocalDeploymentManager::PreStartCheck()
+{
+	const int runtimePort = 5301;
+	// Check for the known runtime port (5301) which could be blocked.
+	if (CheckIfPortIsBound(runtimePort))
+	{
+		// If it exists offer the user the ability to kill it.
+		if (FMessageDialog::Open(EAppMsgType::YesNo, LOCTEXT("KillPortBlockingProcess", "A required port is blocked by another process (potentially by an old deployment). Would you like to kill this process?")) == EAppReturnType::Yes)
+		{
+			const FString NetStatCmd = FString::Printf(TEXT("netstat"));
+			FString Args = TEXT("-n -o -a");
+			FString StdOut;
+			int32 ExitCode;
+			FString StdErr;
+			bool bSuccess = FPlatformProcess::ExecProcess(*NetStatCmd, *Args, &ExitCode, &StdOut, &StdErr);
+
+			if (ExitCode == ExitCodeSuccess && bSuccess)
+			{
+				// Find which line of the output contains the port we're looking for
+				int portIndex = StdOut.Find(FString::Printf(TEXT(":%i"), runtimePort));
+				// Throw away all lines before
+				FString portLine = StdOut.Mid(portIndex);
+				int portLineEnd = portLine.Find("\r\n");
+				// Throw away all lines after
+				portLine = portLine.Mid(0, portLineEnd);
+
+				// Match the PID
+				const FRegexPattern pidMatch(" +[0-9]+$");
+				FRegexMatcher RegMatcher(pidMatch, portLine);
+				if (RegMatcher.FindNext())
+				{
+					int32 pidIndex = RegMatcher.GetMatchBeginning();
+					if (pidIndex >= 0)
+					{
+						// Throw away everything except the PID
+						FString pid = portLine.RightChop(pidIndex);
+						pid = pid.TrimStart();
+						const FString TaskkillCmd = FString::Printf(TEXT("taskkill /F /PID %s"), *pid);
+						Args = "";
```

Comment:
You should be using the args section for args :D
Change the exe to use to just be taskill and then move the args into args.

Reply:
Good point :P